File: application/src/main/java/org/sheedon/common/data/model/IToolbarModel.java
package org.sheedon.common.data.model;

import androidx.databinding.ObservableField;
import androidx.databinding.ObservableInt;
import androidx.lifecycle.MutableLiveData;

/**
 * 导航栏功能
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2021/10/21 1:49 下午
 */
public interface IToolbarModel {

    // 标题
    MutableLiveData<Integer> getTitle();

    // 菜单标题
    ObservableField<String> getMenuTitle();

    // 菜单是否显示
    ObservableInt getMenuVisibility();

    // 返回是否显示
    int getBackVisibility();

    // 设置标题
    void setTitle(int title);

    // 返回按钮触发
    void onBackClick();

    // 菜单按钮触发
    void onMenuClick();
}

File: application/src/main/java/org/sheedon/common/data/model/IToolbarModel.kt
package org.sheedon.common.data.model

import androidx.databinding.ObservableField
import androidx.databinding.ObservableInt
import androidx.lifecycle.MutableLiveData

/**
 * 导航栏功能
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/5 11:39 下午
 */
interface IToolbarModel {

    // 标题
    fun getTitle(): MutableLiveData<Int>

    // 菜单标题
    fun getMenuTitle(): ObservableField<String>

    // 菜单是否显示
    fun getMenuVisibility(): ObservableInt

    // 返回是否显示
    fun getBackVisibility(): Int

    // 设置标题
    fun setTitle(title: Int)

    // 返回按钮触发
    fun onBackClick()

    // 菜单按钮触发
    fun onMenuClick()
}